src/transformers/integrations.py

```diff
@@ -415,6 +379,8 @@ def init_deepspeed(trainer, num_training_steps):
         model=model,
         model_parameters=model_parameters,
         config_params=config,
+        optimizer=trainer.optimizer,
+        lr_scheduler=trainer.lr_scheduler,
```

Comment:
These need to be conditional - only passed if the config didn't have those already.

Reply:
Set it to None if the config has an optimizer. Same thing for lr_scheduler

Reply:
That is a simpler solution!